src/main/java/org/zalando/nakadi/validation/RFC3339DateTimeValidator.java

```diff
@@ -2,40 +2,30 @@
 
 import org.everit.json.schema.FormatValidator;
 
-import java.time.OffsetDateTime;
 import java.time.format.DateTimeParseException;
 import java.util.Optional;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-import static java.time.format.DateTimeFormatter.ISO_OFFSET_DATE_TIME;
-
 public class RFC3339DateTimeValidator implements FormatValidator {
 
-    private final String errorMessage = "must be a valid date-time";
-
-    // Valid offsets are either Z or hh:mm. The format hh:mm:ss is not valid
-    // Pattern is used to catch situations that are not covered by {@link OffsetDateTime#parse(CharSequence)}
-    private final String dateTimeOffsetPattern = "^.*[Tt]\\d{2}:\\d{2}:\\d{2}.*([zZ]|([+-]\\d{2}:\\d{2}))$";
-    private final Pattern pattern = Pattern.compile(dateTimeOffsetPattern);
-    private final Optional<String> error = Optional.of(errorMessage);
+    private static final String DATE_TIME_OFFSET_PATTERN =
+            "^\\d{4}-([0][1-9]|[1][0-2])-[0-3]\\d{1}[Tt][0-2]\\d{1}:[0-5]\\d{1}:[0-5]\\d{1}.?\\d{0,9}([zZ]|([+-]\\d{2}:\\d{2}))$";
```

Comment:
How costly a regexp is? Maybe we could write our own custom validation function that will be even more efficient.

Reply:
If I'm not mistaken - there is a datetime validator that is used in library as well. here we are just checking format. As @SergKam maybe we should test exactly for differences?

Reply:
on local machine it looks like that:
Regex:
1: 849ms
2: 886 ms
3: 2728ms

OffsetDateTime:
1: 6539ms
2: 11627ms
3: 6674ms

Reply:
About more efficient, we could start testing regex approach and approve it later on if required